modules/core/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -312,13 +320,25 @@ public boolean completeUpload(String batchId, String fileIndex, BatchFileInfo fi
         try {
             batch.addFile(fileIndex, blob, blob.getFilename(), blob.getMimeType());
         } catch (NuxeoException e) {
-            amazonS3.deleteObject(bucket, bucketKey);
+            amazonS3.deleteObject(bucket, fileKey);
```

Comment:
It's likely that in the presence of versions this will be incorrect too, the `fileKey` will contain a `@versionid` but we want to do a delete of a specific version (-> parsing and use of `amazonS3.deleteVersion`, see `S3BlobStore.deleteBlob`) ...

Reply:
The `fileKey` here is coming from the client so I don't think we'll ever have a `@versionId`